auth/service.go

```diff
@@ -956,3 +961,122 @@ func DecodeDomainUserID(domainUserID string) (string, string) {
 		return "", ""
 	}
 }
+
+func (svc service) DeleteEntityPolicies(ctx context.Context, entityType, id string) (err error) {
+	switch entityType {
+	case ThingType:
+		// Remove policy of groups
+		req := PolicyReq{
+			SubjectType: GroupType,
+			Object:      id,
+			ObjectType:  ThingType,
+		}
+
+		if err := svc.DeletePolicy(ctx, req); err != nil {
+			return err
+		}
+
+		// Remove policy from domain
+		req.SubjectType = DomainType
+		if err := svc.DeletePolicy(ctx, req); err != nil {
+			return err
+		}
+
+		// Remove policy of users
+		req.SubjectType = UserType
+		if err := svc.DeletePolicy(ctx, req); err != nil {
+			return err
+		}
+
+		return nil
```

Comment:
For `svc.DeletePolicy` only object Type is required field. 
So the below snippet will delete all type of subjects related to thing object
```suggestion
		// Remove all policies related to thing
		req := PolicyReq{
			Object:     id,
			ObjectType: ThingType,
		}

		if err := svc.DeletePolicy(ctx, req); err != nil {
			return err
		}
		return nil
```

Reply:
This doesn't work. I get 
```json
{"time":"2024-03-21T08:36:41.624438869Z","level":"WARN","msg":"Delete entity policies failed to complete successfully","duration":"284.946µs","entity_type":"thing","id":"feae8203-452b-4f7c-a003-76b475e43b56","error":{"error":"malformed entity specification","message":"failed to remove the policies"}}
```